src/settings/LocationLocations/LocationForm/LocationFormContainer.js

```diff
@@ -89,6 +91,10 @@ const LocationFormContainer = ({
 
     saveData(data)
       .then(onSave)
+      .then(() => {
+        queryClient.invalidateQueries(REMOTE_STORAGE_CONFIGURATIONS);
+        queryClient.invalidateQueries(REMOTE_STORAGE_MAPPINGS);
```

Comment:
I'm having a hard time following the logic of LocationFormContainer, but I wonder if this is the right location for these `invalidateQueries` calls. Why do they belong here rather than in `onSave()`?